### PR TITLE
Introduce a Registry class to store the definitions.

### PR DESCRIPTION
**Pull request by jenshaase**
Currently all definitions have been stored in $GLOBALS. This seem to be a bit
of a bad design decision. This is solved with this PR. It introduces a new
Registry to store the definitions.

### 🔖 Changes

* The Registry class was added
* The Emitter have been changed to use the Registry
* All tests have been updated
* All code that used `$GLOBALS` changed to use the Registry

### CODE REVIEW

File: src/php/Lang/Registry.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    public function __construct()
+    {
+        $this->clear();
+    }
+
+    /**
+     * @return self
+     */
+    public static function getInstance()
```
Comment (JesusValera) [inline src/php/Lang/Registry.php:28]: Is there any reason this class is a Singleton?
It couldn't be possible to create from the factory? :)

Comment (JesusValera) [inline src/php/Lang/Registry.php:28]:
```php
public static function getInstance(): self
{
    //...
```
Is the return type `self` not working?

**Review by JesusValera — COMMENTED** [review]
Awesome work! ✨

File: src/php/Lang/Registry.php (hunk 1 above)
Comment (Chemaclass) [reply src/php/Lang/Registry.php:28]: There is no Factory inside the Lang module/directory. This Registry is the replacement for the `$GLOBALS` usage, so for now I think it's way better to use this Singleton instead.

**Review by Chemaclass — COMMENTED** [review]
Nice 👌

**Pull request merged**